# GRN_manuscript Tables

**Table S1. Maize TF mutant alleles isolated from the UniformMu population to test GRN predictions.** Mutant allele transcriptome data obtained, and transcriptome assembly predicted transcript structure included.

**Data File S1**Fasta file of TRINITY *de novo* assembled transcripts for all mutant alleles and control. 

**Table S2. Gene expression values for the 22 TFs in different tissues.** The expression value (CPM) for each of the 22 TFs was assessed based on prior sampling of tissues or developmental stages in B73 (Zhou et al. 2019). Values highlighted indicate the predicted expression level of TF genes in tissues sampled for RNAseq in this study.

**Table S3. Transcription factor differential gene expression data.** Differential expression data averaged across biological replicates (N-rep) in mutant allele (CPM) and W22 control (W22_CPM) samples (log2fc: log2 fold change of mutant to control, lfcSE: log fold change standard error and FDR adjusted p-value included). 

**Table S4. Differentially expressed genes for each mutant allele.** Up- and down-regulated DE genes (log2fc ≥ 1 and FDR adjusted p-value < 0.05) for each mutant allele and those shared by multiple independent alleles per TF in this study.

**Table S5. Significant hypergeometric enrichment of GO terms associated with transcription factor mutant allele differentially expressed genes.** N: differentially expressed genes associated with a GO term-gene IDs listed.

**Table S6. Predicted targets from co-expression based GRN and Y1H methods for all transcription factors in this study.** For TFs with Y1H predictions, total targets (n) represents all B73v4 target genes. For TFs with GRN n1; TF-target predicted in at least 1/45 GRNs and/or GRN n3; TF-target predicted in at least 3/45 GRNs,  predictions, total targets (n) represent a subset of the predicted B73v4 targets that have an annotated W22 gene model and are expressed (CPM ≥ 1 in at least one sample/tissue). 
